src/libraries/System.Private.CoreLib/src/System/IO/Directory.cs

```diff
@@ -271,6 +271,7 @@ public static void Move(string sourceDirName, string destDirName)
             bool sameDirectoryDifferentCase =
                 directoriesAreCaseVariants &&
                 destDirNameFromFullPath.Equals(sourceDirNameFromFullPath, fileSystemSensitivity);
+            bool sourcePathIsFile = FileSystem.FileExists(fullsourceDirName);
```

Comment:
Seems like you want to hit the file system once to get both whether it's an existing directory, and whether it's an existing file.

@stephentoub I can't remember how one does this without pinvoke. If I create a DirectoryInfo and it's a file, it says exists is false. However, if the access time is not default, it is a file that exists. With FileInfo, I can do the same in reverse. Strange behavior and surely there's a better way..?